variables.tf

```diff
@@ -92,6 +92,23 @@ variable "member_cpu_count" {
   }
 }
 
+variable "service_credentials" {
+  description = "A list of service credentials that you want to create for the database"
+  type        = list(string)
+  default     = []
+}
+
+variable "service_credential_roles" {
+  description = "A list of service credentials roles that you want to create for the database, if not provided then role will set as Administrator"
+  type        = list(string)
+  default     = []
+
+  validation {
+    condition     = alltrue([for role in var.service_credential_roles : contains(["Administrator", "Operator", "Viewer", "Editor"], role)])
+    error_message = "Valid values for service_credential_roles are 'Administrator', 'Operator', 'Viewer', and 'Editor'"
+  }
+}
+
```

Comment:
For something as important to keep straight as credential name/role pairing it would be better to have them in a map.
Something like { "name_1": "role_1", "name_2": "role_2" } or { "role_1": ["name_1", "name_2], "role_2": ["name_3"] }
By linking them in this way, it avoids accidental updates to the list altering the pairing.

`service_credentials` would also be better called `service_credentials_names` to avoid confusion with the actual credentials in ibm_resource_key.service_credentials.

Reply:
I have opted for { "name_1": "role_1", "name_2": "role_2" }